core/src/main/scala/org/http4s/UrlForm.scala

```diff
@@ -28,16 +34,17 @@ object UrlForm {
   val empty: UrlForm = new UrlForm(Map.empty)
 
   def apply(values: Map[String, Seq[String]]): UrlForm =
-    // value "" -> Seq() is just noise and it is not maintain during encoding round trip
-    if(values.get("").fold(false)(_.isEmpty)) new UrlForm(values - "")
-    else new UrlForm(values)
+    // value key -> Seq() is just noise and it is not maintain during encoding round trip
+    new UrlForm(values.filter(_._2.nonEmpty))
```

Comment:
Ah.  `"foo"` decodes to `"foo" -> Seq.empty`, `"foo="` decodes to `"foo" -> Seq("")`.